### ISSUE TITLE
UI Listener gobbles singular printf newlines

### ISSUE THREAD

**Issue by catb0t** [factor/factor#1661]
In the `readline-listener`:
```factor
IN: scratchpad "\n" printf

IN: scratchpad "10\n" printf
10
IN: scratchpad "10\n\n" printf
10

IN: scratchpad
```
Works as expected.

In the GUI:
```factor
IN: scratchpad "\n" printf
IN: scratchpad "10\n" printf
10IN: scratchpad "10\n\n" printf
10
IN: scratchpad
```

Is that intentional?

**Comment by mrjbq7** [comment]
Related to handling of ``write``:

```factor
IN: scratchpad "\n" write
IN: scratchpad "10\n" write
10IN: scratchpad "10\n\n" write
10
IN: scratchpad
```

Maybe related to my change in ef2ae9c360721eab10bad9c7bee9161ed8b5255e, previously ``string-lines`` did this:

```factor
IN: scratchpad "\n" string-lines
{ "" "" }
```

Now it does this:

```factor
IN: scratchpad "\n" string-lines
{ "" }
```

I think it's better but ``pane-lines`` (https://github.com/factor/factor/blob/master/basis/ui/gadgets/panes/panes.factor#L102) uses ``string-lines`` and maybe it needs my "hack" that I used for documents (https://github.com/factor/factor/blob/master/basis/documents/documents.factor#L113) or a refactoring to do the right thing with the new behavior...

**Comment by mrjbq7** [comment]
Fixed in e8a9582400ac06a11d35830ead53be1130d0d355.

We now have two uses of the old ``string-lines`` word (that is kept in ``documents.private:split-lines``).  Perhaps we should look at whether that should exist somewhere more public.
